code/nrf-connect/samples/zigbee/src/main.c

```diff
@@ -129,14 +133,37 @@ void zboss_signal_handler(zb_bufid_t bufid) {
         k_timer_stop(&led_flashing_timer);
         prst_led_off();
       }
-      break;
     }
-    case ZB_ZDO_SIGNAL_SKIP_STARTUP:
-    case ZB_ZDO_SIGNAL_LEAVE: {
+    case ZB_BDB_SIGNAL_DEVICE_FIRST_START:
+      joining_signal_received = true;
+      break;
+    case ZB_ZDO_SIGNAL_LEAVE:
+      if (status == RET_OK) {
+        zb_zdo_signal_leave_params_t *leave_params = ZB_ZDO_SIGNAL_GET_PARAMS(sig_hndler, zb_zdo_signal_leave_params_t);
+        LOG_INF("Network left (leave type: %d)", leave_params->leave_type);
+
+        /* Set joining_signal_received to false so broken rejoin procedure can be detected correctly. */
+        if (leave_params->leave_type == ZB_NWK_LEAVE_TYPE_REJOIN) {
+          joining_signal_received = false;
+        }
+      }
+    case ZB_ZDO_SIGNAL_SKIP_STARTUP: {
+      stack_initialised = true;
       LOG_DBG("Will restart flashing");
       k_timer_start(&led_flashing_timer, K_NO_WAIT, K_SECONDS(1));
       break;
     }
+    case ZB_NLME_STATUS_INDICATION: {
+      zb_zdo_signal_nlme_status_indication_params_t *nlme_status_ind =
+          ZB_ZDO_SIGNAL_GET_PARAMS(sig_hndler, zb_zdo_signal_nlme_status_indication_params_t);
+      if (nlme_status_ind->nlme_status.status == ZB_NWK_COMMAND_STATUS_PARENT_LINK_FAILURE) {
+        /* Check for broken rejoin procedure and restart the device to recover. */
```

Comment:
It would be great to leave a comment linking this workaround to [Known issues](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/known_issues.html). Hopefully we can remove this code in a future SDK version.

Reply:
I've updated the comment with details and a link.